### PR TITLE
Custom options handler - fixed merging options.

### LINKED ISSUE #11: The `body` option from `got` does not work at all
This code does not send the body at all. It also does not work with the `json` option. I haven't tested with `form`, but I assume it will be the same.

```js
const got = require('got-scraping');

got({
    url: 'https://en2ydmzx2ein.x.pipedream.net/',
    body: '{ foo: "bar" }',
    method: 'POST',
    ciphers: undefined,
}).then((res) => {
    console.dir(res);
});
```

If you replace the `require` with `got`, it works fine. The above code also does not fail when the `method: 'POST'` is removed. Whereas in `got` it throws

```
RequestError: The `GET` method cannot be used with a body
```

### PR DESCRIPTION
**Pull request by petrpatek**
closes #11

### CODE REVIEW

**Comment by mnmkng** [comment]
Is this ready for review? The tests are failing and it's still in draft.

File: test/main.test.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ const gotScraping = require('../src');
 
 const { startDummyServer } = require('./helpers/dummy-server');
 
+process.env.APIFY_PROXY_PASSWORD = 'rgC8JJ8NcrDnDdwsxDqWz7jKS';
```
Comment (mnmkng) [inline test/main.test.js:7]: Is this a real proxy password? If it is, please invalidate it.

File: src/handlers/custom-options.js [hunk 2]
```diff
@@ -9,17 +11,23 @@ function customOptionsHandler(options, next) {
...
         ...context,
         proxyUrl,
         headerGeneratorOptions,
         useHeaderGenerator,
     };
 
-    return next(gotOptions);
+    delete options.proxyUrl;
+    delete options.headerGeneratorOptions;
+    delete options.useHeaderGenerator;
```
Comment (mnmkng) [inline src/handlers/custom-options.js:26]: Aren't you deleting this from the object the user provided?

File: test/main.test.js [hunk 3]
```diff
@@ -91,6 +92,37 @@ describe('GotScraping', () => {
...
+
+        const response = await gotScraping({
+            responseType: 'json',
+            url: `http://localhost:${port}/jsonPost`,
+            json: body,
+            method: 'POST',
+        });
+
+        expect(response.statusCode).toBe(200);
+        expect(response.body).toMatchObject(body);
```
Comment (mnmkng) [inline test/main.test.js:106]: I think `.toMatchObject` allows partial matches. `.toEqual` would be better here.

File: test/main.test.js [hunk 4]
```diff
@@ -91,6 +92,37 @@ describe('GotScraping', () => {
...
+            body: JSON.stringify(body),
+            responseType: 'json',
+            method: 'POST',
+            headers: {
+                'content-type': 'application/json; charset=UTF-8',
+            },
+        });
+
+        expect(response.statusCode).toBe(200);
+        expect(response.body).toMatchObject(body);
```
Comment (mnmkng) [inline test/main.test.js:123]: dtto

**Review by mnmkng — CHANGES_REQUESTED** [review]

File: src/handlers/custom-options.js (hunk 2 above)
Comment (petrpatek) [reply src/handlers/custom-options.js:26]: It should be already a new object provided by got. It is already merged with the defaults.

File: test/main.test.js (hunk 1 above)
Comment (petrpatek) [reply test/main.test.js:7]: Oh, done. 🤦

**Review by mnmkng — APPROVED** [review]

**Pull request merged**